src/System/Net/Benchmarks/Apps/Socket/Client/Program.cs

```diff
@@ -0,0 +1,247 @@
+﻿using System.Diagnostics;
+using System.Net;
+using System.Net.Benchmarks;
+using System.Net.Benchmarks.SocketBenchmark.Client;
+using System.Net.Benchmarks.SocketBenchmark.Shared;
+using System.Net.Sockets;
+
+return await SocketBenchmarkClient.RunAsync(args).ConfigureAwait(false);
+
+internal class SocketBenchmarkClient : BenchmarkClient<SocketClientOptions>
```

Comment:
I was thinking, and I still believe we'd need to separate these benchmarks (which are technically `NetworkStream` benchmarks) from the to-be-added pure-Socket-API benchmarks. Otherwise it would be hard to navigate within the benchmark class -- this one is big enough already.

And creating a socket is technically a two-liner (`var socket = new Socket(...); await socket.ConnectAsync(...);`) and it can/should be potentially different in new benchmarks so we can maybe set some socket options based on passed parameters. So IMO there's not really much to reuse? Or am I missing something?

So, currently the proposal is to rename
`SocketBenchmark` -> `NetworkStreamBenchmark`,
`src/System/Net/Benchmarks/Apps/Socket/` -> `src/System/Net/Benchmarks/Apps/Streaming/`,
and maybe even
`src/System/Net/Benchmarks/Apps/Tls/` -> `src/System/Net/Benchmarks/Apps/Streaming/Tls/`

WDYT?

cc @rzikm for additional opinions

Reply:
> I was thinking, and I still believe we'd need to separate these benchmarks (which are technically NetworkStream benchmarks) from the to-be-added pure-Socket-API benchmarks. Otherwise it would be hard to navigate within the benchmark class -- this one is big enough already.

After thinking a while on my own, I totally agree with this. I'm planning to change this as `NetworkStreamBenchmark` and add another Benchmark for pure Socket APIs, because we can extend those Socket Benchmarks with UDP and possibly(?) with other protocols as well in the future.

> And creating a socket is technically a two-liner (var socket = new Socket(...); await socket.ConnectAsync(...);) and it can/should be potentially different in new benchmarks so we can maybe set some socket options based on passed parameters. So IMO there's not really much to reuse? Or am I missing something?

I also think that we don't have much to reuse here and it is highly possible that some benchmarks can customize their usages of socket while creating connection (even during connection?), maybe we can create a `ConnectCallback` or something like that as we have in `SocketsHttpHandler`. (This is what I'm thinking from top of my head)

> So, currently the proposal is to rename
SocketBenchmark -> NetworkStreamBenchmark,
src/System/Net/Benchmarks/Apps/Socket/ -> src/System/Net/Benchmarks/Apps/Streaming/,
and maybe even
src/System/Net/Benchmarks/Apps/Tls/ -> src/System/Net/Benchmarks/Apps/Streaming/Tls/

The only question that comes in my mind is, should we keep QUIC under Tls folder here. Technically, it's a stream protocol as well but do we want to keep it as separated? Or you don't have any problem with moving it with Tls?